src/azure-cli/azure/cli/command_modules/acr/agentpool.py

```diff
@@ -0,0 +1,129 @@
+    get_registry_by_name,
+    validate_managed_registry
+)
+
+
+DEFAULT_COUNT = 1
+DEFAULT_TIER = 'S1'
+DEFAULT_OS = 'Linux'
+
+logger = get_logger(__name__)
+
+
+def acr_agentpool_create(cmd,
+                         client,
+                         agent_pool_name,
+                         registry_name,
+                         resource_group_name=None,
+                         count=DEFAULT_COUNT,
+                         tier=DEFAULT_TIER,
+                         os_type=DEFAULT_OS,
+                         no_wait=False,
+                         subnet_id=None):
+
+    registry, resource_group_name = get_registry_by_name(
+        cmd.cli_ctx, registry_name, resource_group_name)
+
+    AgentPool = cmd.get_models('AgentPool')
+
+    agentpool_create_parameters = AgentPool(
+        location=registry.location,
+        count=count,
+        tier=tier.upper(),
+        os=os_type,
+        virtual_network_subnet_resource_id=subnet_id
+    )
+
+    try:
+        response = client.create(resource_group_name=resource_group_name,
+                                 registry_name=registry_name,
+                                 agent_pool_name=agent_pool_name,
+                                 agent_pool=agentpool_create_parameters)
+        if no_wait:
+            return client.get(resource_group_name, registry_name, agent_pool_name)
+        return response
+    except ValidationError as e:
+        raise CLIError(e)
+
+
+def acr_agentpool_update(cmd,
+                         client,
+                         agent_pool_name,
+                         registry_name,
+                         resource_group_name=None,
+                         no_wait=False,
+                         count=None):
+
+    _, resource_group_name = validate_managed_registry(
+        cmd, registry_name, resource_group_name)
+
+    try:
+        response = client.update(resource_group_name=resource_group_name,
+                                 registry_name=registry_name,
+                                 agent_pool_name=agent_pool_name,
+                                 count=count)
+        if no_wait:
+            return client.get(resource_group_name, registry_name, agent_pool_name)
+        return response
+    except ValidationError as e:
+        raise CLIError(e)
+
+
+def acr_agentpool_delete(cmd,
+                         client,
+                         agent_pool_name,
+                         registry_name,
+                         no_wait=False,
+                         resource_group_name=None):
+
+    _, resource_group_name = validate_managed_registry(
+        cmd, registry_name, resource_group_name)
+
+    try:
+        response = client.delete(resource_group_name=resource_group_name,
+                                 registry_name=registry_name,
+                                 agent_pool_name=agent_pool_name)
+
+        if no_wait:
+            return client.get(resource_group_name, registry_name, agent_pool_name)
```

Comment:
in theory, the delete can return 200 which indicates the agent pool has been successfully removed. if you call client.get, it will get 404 error. Is it why you want to wrap response in the longrunningoperation initially?

Reply:
Ok, I think you can just do something like 
```
        if no_wait:
            logger.warning("Started to delete the agent pool: %s", response.status())
            return
```